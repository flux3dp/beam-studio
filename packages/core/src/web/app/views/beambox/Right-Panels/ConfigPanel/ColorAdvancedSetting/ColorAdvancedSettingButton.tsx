import React, { useContext } from 'react';

import ConfigPanelIcons from '@core/app/icons/config-panel/ConfigPanelIcons';

import styles from '../Block.module.scss';
import ConfigPanelContext from '../ConfigPanelContext';

import { showColorAdvancedSetting } from './utils';

const ColorAdvancedSettingButton = () => {
  const { selectedLayers } = useContext(ConfigPanelContext);

  return (
    <>
      <div className={styles.panel}>
        <span className={styles.title} onClick={() => showColorAdvancedSetting(selectedLayers)}>
          Color Advanced Setting
          <span className={styles.icon}>
            <ConfigPanelIcons.ColorAdjustment />
          </span>
        </span>
      </div>
    </>
  );
};

export default ColorAdvancedSettingButton;
