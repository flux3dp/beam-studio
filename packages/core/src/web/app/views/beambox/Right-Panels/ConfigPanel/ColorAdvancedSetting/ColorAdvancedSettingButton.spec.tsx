import React, { createContext } from 'react';

import { fireEvent, render } from '@testing-library/react';

const mockShowColorAdvancedSetting = jest.fn();

jest.mock('./utils', () => ({
  showColorAdvancedSetting: (...args) => mockShowColorAdvancedSetting(...args),
}));

jest.mock('../ConfigPanelContext', () => createContext({ selectedLayers: ['layer1', 'layer2'] }));

import ColorAdvancedSettingButton from './ColorAdvancedSettingButton';

describe('ColorAdvancedSettingButton', () => {
  it('should renders correctly', () => {
    const { container } = render(<ColorAdvancedSettingButton />);

    expect(container).toMatchSnapshot();
  });

  it('should call showColorAdvancedSetting when clicked', () => {
    const { container } = render(<ColorAdvancedSettingButton />);
    const title = container.querySelector('.title');

    fireEvent.click(title);
    expect(mockShowColorAdvancedSetting).toHaveBeenCalled();
  });
});
