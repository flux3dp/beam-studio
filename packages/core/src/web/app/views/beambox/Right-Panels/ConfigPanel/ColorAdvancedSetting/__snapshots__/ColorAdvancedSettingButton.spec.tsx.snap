// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorAdvancedSettingButton should renders correctly 1`] = `
<div>
  <div
    class="panel"
  >
    <span
      class="title"
    >
      Color Advanced Setting
      <span
        class="icon"
      >
        <svgr-url />
      </span>
    </span>
  </div>
</div>
`;
